### PR TITLE
:sparkles: If enabled is set to false, the MachineHealthCheckClass field should not be set.

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes-sigs/cluster-api/issues/8721

/area clusterclass

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

**Comment by chaunceyjiang** [comment]
Hi @killianmuldoon @sbueringer @fabriziopandini Please take a look. Thanks~

**Comment by fabriziopandini** [comment]
q: Have we considered scenarios like someone who wants to temporarily disable MHC, e.g. for troubleshooting? 
Now this is possible just setting enabled to false and then back to true; after this change, on top of that the user will be forced to delete the MHC configuration, keep it noted somewhere else, and restore it later.
Might be this should be a warning, not an error...

**Review by killianmuldoon — COMMENTED** [review]
@chaunceyjiang there's an example of adding a webhook warning here: https://github.com/kubernetes-sigs/cluster-api/pull/8746/files

I think @fabriziopandini is right and that's the approach this PR should take.

**Comment by vincepri** [comment]
Agree with @fabriziopandini, it should be a warning, but the configuration should not be lost

File: internal/webhooks/cluster_test.go [hunk 1]
```diff
@@ -1481,6 +1481,34 @@ func TestClusterTopologyValidationWithClient(t *testing.T) {
 			classReconciled: true,
 			wantErr:         true,
 		},
+		{
+			name: "Reject a cluster that has MHC disabled for machine deployment but is MHC definition in cluster topology and ClusterClass",
```
Comment (vincepri) [inline internal/webhooks/cluster_test.go:1485]: Update?

File: internal/webhooks/cluster.go [hunk 2]
```diff
@@ -419,13 +421,20 @@ func validateMachineHealthChecks(cluster *clusterv1.Cluster, clusterClass *clust
...
+				if cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() && clusterClass.Spec.ControlPlane.MachineHealthCheck == nil {
+					allErrs = append(allErrs, field.Forbidden(
+						fldPath.Child("enable"),
+						fmt.Sprintf("cannot be set to %t as MachineHealthCheck definition is not available in the Cluster topology or the ClusterClass", *cluster.Spec.Topology.ControlPlane.MachineHealthCheck.Enable),
+					))
+				}
+			} else {
+				// If enabled is set to false, the MachineHealthCheckClass field should not be set.
+				if !cluster.Spec.Topology.ControlPlane.MachineHealthCheck.MachineHealthCheckClass.IsZero() {
+					allWarnings = append(allWarnings, "if the MachineHealthCheck enabled is set to false, the MachineHealthCheckClass field should not be set.")
```
Comment (vincepri) [inline internal/webhooks/cluster.go:436]:
```suggestion
					allWarnings = append(allWarnings, "MachineHealthCheck.Enable is set to false, class options won't be set")
```

or something similar

Comment (chaunceyjiang) [reply internal/webhooks/cluster.go]: Updated.

**Pull request closed without merging**